sentry-android-core/src/main/java/io/sentry/android/core/AndroidOptionsInitializer.java

```diff
@@ -58,11 +59,14 @@ private static void setDefaultInApp(Context context, SentryOptions options) {
     options.addInAppExclude("kotlin.");
   }
 
+  @SuppressWarnings("FutureReturnValueIgnored")
   private static void initializeCacheDirs(Context context, SentryOptions options) {
-    File cacheDir = new File(context.getCacheDir(), "sentry");
-    cacheDir.mkdirs();
-    options.setCacheDirPath(cacheDir.getAbsolutePath());
-    new File(options.getOutboxPath()).mkdirs();
+    String cacheDirPath = context.getCacheDir().getAbsolutePath() + File.separator + "sentry";
+    options.setCacheDirPath(cacheDirPath);
+
+    // create files async to avoid IO on main thread.
+    new AsyncFileCreator().createFile(cacheDirPath);
+    new AsyncFileCreator().createFile(options.getOutboxPath());
```

Comment:
This will leave 2 threads lingering forever and eating up resources needlessly.

Reply:
@metlos ups, you are right, missing `shutdown`